src/cheri_insts.sail

```diff
@@ -67,20 +83,41 @@ function clause execute (CGetType(rd, cs1)) =
   RETIRE_SUCCESS
 }
 
+/*!
+ * Integer register *rd* is set equal to the **base** field of capability
+ * register *cs1*.
+ */
 function clause execute (CGetBase(rd, cs1)) =
 {
   let capVal = C(cs1);
   X(rd) = getCapBaseBits(capVal);
   RETIRE_SUCCESS
 }
 
+/*!
+ * Integer register *rd* is set equal to the **offset** field of capability
+ * register *cs1*.
+ */
 function clause execute (CGetOffset(rd, cs1)) =
 {
   let capVal = C(cs1);
   X(rd) = getCapOffsetBits(capVal);
   RETIRE_SUCCESS
 }
 
+/*
+ * Note: We have to use [{xlen}][xlen] instead of [xlen] to avoid \lstinline{}
+ * inside math mode.
+ */
+/*!
+ * Integer register *rd* is set equal to the **length** field of capability
+ * register *cs1*.
+ *
+ * ## Notes
+ * Due to the compressed representation of capabilities, the actual length
+ * of capabilities can be $2^{[{xlen}][xlen]}$; [CGetLen] will return the maximum
+ * value of $2^{[{xlen}][xlen]}-1$ in this case.
```

Comment:
I'm unsure how to deal with notes. Some of them make sense to keep in Sail, but some of the MIPS ones refer to uarch things and don't make so much sense there.

Reply:
I dropped some of the notes that don't make sense for RISC-V, but I felt like this one might make sense here?

Reply:
The issue though is what to do if we decide we do have uarch-related notes we want to include.

Reply:
How about: uarch notes that affect the architectural instruction semantics go into the sail, everything else into the uarch chapter of the ISA ref?

Reply:
That seems sensible; we'll see whether there are any notes that don't fall into either category nicely.